ip_ranges.go

```diff
@@ -0,0 +1,45 @@
+package netaddr
+
+import (
+	"fmt"
+	"net"
+)
+
+// IPRange range of ips not necessarily aligned to a power of 2
+type IPRange struct {
+	First, Last net.IP
```

Comment:
I wonder how wise it is to expose these directly. I could create this with all kinds of strange values. I could mix ipv4 and ipv6. I could create an invalid range when Last < First, I could put in a couple of `net.IP{}`s. What are your thoughts?

In contrast to this, my proposed API exposed First and Last as methods.

Reply:
If this were java, I'd be all about that coding protection stuff, but here I was going for KIS. 

Also, in the other part about `LessThan` I introduced a non-canonical ordering of addresses that traverses both ipv4 and ipv6.  I feel like it's useful and not worth introducing an error return type for that method.

For the range intersection algorithm, it uses that ordering for its intersections  so if first=ipv4 last could be ipv6 and things would span both.

I'm not sure if this is invalid or not, but it does make it easy to see how an ipv4 range does not intersect with an ipv6 range